presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -326,7 +326,9 @@ public FSDataOutputStream create(Path path, FsPermission permission, boolean ove
             throw new IOException("File already exists:" + path);
         }
 
-        createDirectories(stagingDirectory.toPath());
+        if (!stagingDirectory.exists()) {
+            createDirectories(stagingDirectory.toPath());
+        }
```

Comment:
For now I think I will add a new test that does the former (sets the staging directory to a temporary *file* rather than a directory then expects an appropriate error message) but will not add a test for the latter unless you think it necessary.